### PR TITLE
Redid the integration tests using serverless-offline

### PR DESCRIPTION
**Pull request by palfrey**
As part of looking at #280, I couldn't figure out how to get the integration tests running without actually deploying things to AWS. This PR redoes them with `serverless-offline`.

### CODE REVIEW

File: test/integration/test.js [hunk 1]
```diff
@@ -48,14 +60,21 @@ runtimes.forEach(runtime => {
...
-    before(async function() {
-      this.timeout(10000);
-      const info = await run('info');
-      endpoints = await getEndpoints(info.split('\r?\n'));
+    after(async function() {
+      let done = false;
+      this.subprocess.on('close', () => {
+        done = true;
+      })      
+      kill(this.subprocess.pid);
```
Comment (palfrey) [inline test/integration/test.js:74]: Tried doing this with `this.subprocess.signal('SIGKILL')`, etc but no luck. [tree-kill](https://www.npmjs.com/package/tree-kill) OTOH works fine!

File: test/integration/test.js [hunk 2]
```diff
@@ -139,7 +158,8 @@ runtimes.forEach(runtime => {
         .expect('Content-Type', /json/);
     });
 
-    it('root', () => {
+    // FIXME: Broken currently https://github.com/dougmoscrop/serverless-http/issues/270
+    it.skip('root', () => {
```
Comment (palfrey) [inline test/integration/test.js:162]: See https://github.com/dougmoscrop/serverless-http/issues/270

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "pretest": "tsc --strict --skipLibCheck --noEmit test/typecheck.ts",
     "test": "nyc mocha",
     "posttest": "eslint lib test",
-    "test:integration": "mocha test/integration/test.js",
+    "test:integration": "mocha test/integration/test.js --exit",
```
Comment (palfrey) [inline package.json:21]: Not sure what's hanging around given the process kills, but doesn't work without this.

**Comment by dougmoscrop** [comment]
Interesting, I'm a fan of using the actual AWS service in case there's drift between serverless-offline, but I also see the value in being able to use it - do you think we could support both?  `npm run integration:offline`, `npm run integration:aws` etc?

**Comment by palfrey** [comment]
> Interesting, I'm a fan of using the actual AWS service in case there's drift between serverless-offline, but I also see the value in being able to use it - do you think we could support both? `npm run integration:offline`, `npm run integration:aws` etc?

Added. I've also plugged the offline integration into the new Github actions work.

**Comment by dougmoscrop** [comment]
@palfrey do you mind rebasing?

**Comment by palfrey** [comment]
This all got merged as part of https://github.com/dougmoscrop/serverless-http/pull/286

**Pull request closed without merging**